src/NoUnused/Exports.elm

```diff
@@ -547,6 +553,46 @@ fromProjectToModule =
         |> Rule.withModuleDocumentation
 
 
+declarationToTopLevelExpose : Range -> Declaration -> Maybe TopLevelExpose
+declarationToTopLevelExpose range declaration =
+    case declaration of
+        Declaration.FunctionDeclaration function ->
+            function.declaration
+                |> Node.value
+                |> .name
+                |> Node.value
+                |> Exposing.FunctionExpose
+                |> Just
+
+        Declaration.AliasDeclaration typeAlias ->
+            typeAlias.name
+                |> Node.value
+                |> Exposing.TypeOrAliasExpose
+                |> Just
+
+        Declaration.CustomTypeDeclaration type_ ->
+            Just <|
+                Exposing.TypeExpose
+                    { name = Node.value type_.name
+                    , open = Just range
+                    }
+
+        Declaration.PortDeclaration signature ->
+            signature.name
+                |> Node.value
+                |> Exposing.FunctionExpose
+                |> Just
+
+        Declaration.InfixDeclaration infix ->
+            infix.operator
+                |> Node.value
+                |> Exposing.InfixExpose
+                |> Just
+
+        Declaration.Destructuring _ _ ->
```

Comment:
It would in theory be possible to implement (some of) this, but it didn't seem worth it.

Reply:
This AST node is not possible actually. I aim to remove it in a future major version of `elm-syntax`. Until then we have to live with it... :upside_down_face: 
So the code here is fine :+1: